internal/npm_install/generate_pnp_build_files.ts

```diff
@@ -1,44 +1,47 @@
-import * as fs from "fs";
+import { promises as fsp, existsSync, mkdirSync } from "fs";
 import * as path from "path";
 
-const workspacePath = process.argv.slice(2)[0];
-
-function mkdirp(p: string) {
-  if (!fs.existsSync(p)) {
-    mkdirp(path.dirname(p));
-    fs.mkdirSync(p);
-  }
-}
-
 function writeFileSync(p: string, content: string) {
-  mkdirp(path.dirname(p));
-  fs.writeFileSync(p, content);
+  if (!existsSync(path.dirname(p))) {
+    mkdirSync(path.dirname(p), { recursive: true });
+  }
+  return fsp.writeFile(p, content);
 }
 
-function createBuildFile(pkg: string) {
+function createBuildFile(packagePath: string) {
   const contents = `package(default_visibility = ["//visibility:public"])
 load("@build_bazel_rules_nodejs//internal/npm_install:node_module_library.bzl", "node_module_library")
 
 filegroup(
-    name = "${pkg}",
+    name = "${packagePath}",
     srcs = [] # we dont care about the actual content, we want a list of dependencies to modify the pnp file around
-)`;
+)
+`;
 
-  writeFileSync(`${pkg}/BUILD.bazel`, contents);
+  return writeFileSync(`${packagePath}/BUILD.bazel`, contents);
```

Comment:
writeFile is not synchronous anymore which makes it return a promise and thus also makes `createBuildFile` async.

Reply:
this is intentional, I should be dealing with all the async stuff with `Promise.all` though I may have missed something